### PR TITLE
[SPARK-12819] Deprecate TaskContext.isRunningLocally()

### PR DESCRIPTION
**Pull request by JoshRosen**
We've already removed local execution but didn't deprecate `TaskContext.isRunningLocally()`; we should deprecate it for 2.0.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/TaskContext.scala [hunk 1]
```diff
@@ -96,12 +96,6 @@ abstract class TaskContext extends Serializable {
   def isInterrupted(): Boolean
 
   /**
-   * Returns true if the task is running locally in the driver program.
-   * @return
-   */
-  def isRunningLocally(): Boolean
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/TaskContext.scala]: I'd keep the method here... feel free to deprecate.

Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/TaskContext.scala]: Will do.

**Comment by JoshRosen** [comment]
Updated.

File: core/src/main/scala/org/apache/spark/TaskContext.scala [hunk 2]
```diff
@@ -97,8 +97,9 @@ abstract class TaskContext extends Serializable {
 
   /**
    * Returns true if the task is running locally in the driver program.
-   * @return
+   * @return false
    */
+  @deprecated("Local execution was removed,so this always returns false", "2.0.0")
```
Comment (rxin) [inline core/src/main/scala/org/apache/spark/TaskContext.scala]: nit: add a space after comma

**Comment by rxin** [comment]
LGTM

**Comment by rxin** [comment]
I've merged this.

**Pull request closed without merging**